Close #5837: Document reconcile period for Ansible in advanced options (#5848)
operator-framework/operator-sdk: website/content/en/docs/building-operators/ansible/reference/advanced_options.md

2. `Tasks` - Only Ansible Tasks will be outputted.
3. `Everything` - All info logs and all tasks will be outputted.

If you want more control over the logs that are outputted, consider using the [Zap Logger][Zap-Logger] and [verbosity annotations][verbosity-annotations] in tandem with the `--ansible-log-events` CLI flag.

> ## `ansible.sdk.operatorframework.io/reconcile-period` Custom Resource Annoation

Reviewer: ## `ansible.sdk.operatorframework.io/reconcile-period` Custom Resource Annotation

> While running a Ansible-based operator, the reconcile-period can be specified through the custom resource's annotations under the `ansible.sdk.operatorframework.io/reconcile-period` key. 

Reviewer 2: ```suggestion
You can specify the reconcile period for an Ansible Operator by editing the `ansible.sdk.operatorframework.io/reconcile-period` key in the custom resource annotation file. 
```
Would it help the user to have the full filepath here? Like: "... in the `watches.yaml` custom resource annotation file."

Author: "Editing" suggests that this field is already present in the custom resource when the operator gets scaffolded. However, a user would have to manually add this field to use this feature.

I think mentioning custom resource is sufficient since it is a major component of the operator and authors usually know which file (YAML formatted) this corresponds to. Annotations are a subsection of this file.

Reviewer 2: WDYT of the following:
"You can specify the reconcile period for an Ansible Operator by adding the `ansible.sdk.operatorframework.io/reconcile-period` key to the custom resource annotations."

> While running a Ansible-based operator, the reconcile-period can be specified through the custom resource's annotations under the `ansible.sdk.operatorframework.io/reconcile-period` key. 

Reviewer: While running an Ansible-based operator, the reconcile-period can be specified through the custom resource's annotations under the `ansible.sdk.operatorframework.io/reconcile-period` key.

> This feature guarantees that an operator will get reconciled, at minimum, in the specified interval of time. In other words, it ensures that the cluster will not go longer
> than the specified reconcile-period without being reconciled. However, the cluster may be reconciled at any moment if there are changes detected in the desired state.

Reviewer 2: ```suggestion
This feature specifies the maximum interval an Operator will get reconciled. If changes are detected in the desired state, the cluster may be reconciled sooner than the specified interval.
```

The reconcile period can be specified in the custom resource's annotations in the following manner: 

> ```sh

Reviewer 2: Should this be `sh` or `yaml`?

Author: Should be yaml, good call out

...
metadata:
  name: memcached-sample
  annotations:
    ansible.sdk.operatorframework.io/reconcile-period: 5s
...
```

The value that is present under this key must be in the h/m/s format. For example, 1h2m4s, 3m0s, 4s are all valid values, but 1x3m9s is invalid. 

> **NOTE**: This is just one way of specifying the reconcile period for Ansible-based operators. There are two other ways: using the `--reconcile-period` command-line flag and under the 'reconcilePeriod' key in the watches.yaml file. If these three methods are used simultaneously to specify reconcile period (which they should not be), the order of precedence is as follows: 

Reviewer 2: ```suggestion
**NOTE**: Alternatively, you can specify the reconcile period for Ansible-based Operators in the following ways:
- Running the `--reconcile-period` command-line flag 
- Editing the 'reconcilePeriod' key in the `watches.yaml` file. 

You should not use all three methods to specify a single reconcile period. If all three methods are used simultaneously, the order of precedence is as follows: 
```
I am not totally sure if I caught the right meaning in the last section.

> Custom Resource Annotations > watches.yaml > command-line flag.

Reviewer 2: ```suggestion
Custom resource annotations > `watches.yaml` file > command-line flag.
```

If you want more control over the logs that are outputted, consider using the [Zap Logger][Zap-Logger] and [verbosity annotations][verbosity-annotations] in tandem with the `--ansible-log-events` CLI flag.

## `ansible.sdk.operatorframework.io/reconcile-period` Custom Resource Annotation

You can specify the reconcile period for an Ansible Operator by using the `ansible.sdk.operatorframework.io/reconcile-period` key in the custom resource annotation file. 
This feature specifies the maximum interval in which a cluster will get reconciled. If changes are detected in the desired state, the cluster may be reconciled sooner than the specified interval.

The reconcile period can be specified in the custom resource's annotations in the following manner: 

```yaml
...
metadata:
  name: memcached-sample
  annotations:
    ansible.sdk.operatorframework.io/reconcile-period: 5s
...
```

The key only accepts a value in the `h/m/s` format, such as `1h2m4s`, `3m0s`, or `4s`. Values such as `1x3m9s` are invalid.

**NOTE**: Alternatively, you can specify the reconcile period for Ansible-based Operators in the following ways:
- Using the `--reconcile-period` command-line flag 

> - Using the 'reconcilePeriod' key in the `watches.yaml` file. 

Reviewer 2: ```suggestion
- Using the 'reconcilePeriod' key in the `watches.yaml` file
```
